src/parser/SparqlParser.cpp

```diff
@@ -275,6 +275,59 @@ void SparqlParser::parseWhere(ParsedQuery* query,
       // Recursively call parseWhere to parse the optional part.
       parseWhere(query, &child);
       _lexer.accept(".");
+    } else if (_lexer.accept("bind")) {
+      _lexer.expect("(");
+      std::string inVar;
+      bool rename = false;
+      bool isString = true;
+      bool isSum = false;
+      std::string inVar2;
+      int64_t val = 0;
+      if (_lexer.accept(SparqlToken::Type::VARIABLE)) {
+        rename = true;
+        inVar = _lexer.current().raw;
+        if (_lexer.accept(SparqlToken::Type::SYMBOL)) {
+          if (_lexer.current().raw != "+") {
+            throw std::runtime_error(
+                "A sum of two variables is currently the only"
+                "supported arithmetic bind expression, but symbol after first "
+                "variable was " +
+                _lexer.current().raw);
+          }
+          isSum = true;
+          _lexer.expect(SparqlToken::Type::VARIABLE);
+          inVar2 = _lexer.current().raw;
+        }
+      } else if (_lexer.accept(SparqlToken::Type::RDFLITERAL)) {
+        inVar = parseLiteral(_lexer.current().raw, true);
```

Comment:
// The "true" says that the whole string is a literal (with "false", there could be more stuff after the literal).